src/tests/test_piranha_java.rs

```diff
@@ -11,174 +11,155 @@ Copyright (c) 2022 Uber Technologies, Inc.
  limitations under the License.
 */
 
-use crate::models::{default_configs::JAVA, piranha_arguments::PiranhaArgumentsBuilder};
-
 use super::{
-  get_piranha_arguments_for_test, get_piranha_arguments_for_test_with_substitutions, initialize,
-  run_match_test, run_rewrite_test,
+  check_result, copy_folder, create_match_test, create_rewrite_test, initialize, substitutions,
 };
-
-static LANGUAGE: &str = JAVA;
-
-#[test]
-fn test_java_scenarios_treated_ff1() {
-  initialize();
-
-  let relative_path_to_tests = &format!("{}/{}/{}", JAVA, "feature_flag_system_1", "treated");
-
-  let substitutions = vec![
-    vec!["stale_flag_name".to_string(), "STALE_FLAG".to_string()],
-    vec!["treated".to_string(), "true".to_string()],
-    vec!["treated_complement".to_string(), "false".to_string()],
-    vec!["namespace".to_string(), "some_long_name".to_string()],
-  ];
-
-  let piranha_argument =
-    get_piranha_arguments_for_test_with_substitutions(relative_path_to_tests, JAVA, substitutions);
-
-  run_rewrite_test(piranha_argument, 2, relative_path_to_tests);
+use crate::{
+  execute_piranha,
+  models::{
+    default_configs::JAVA,
+    piranha_arguments::{PiranhaArguments, PiranhaArgumentsBuilder},
+  },
+};
+use std::path::Path;
+use tempdir::TempDir;
+
+fn feature_flag_system_1_treated() -> PiranhaArguments {
+  PiranhaArguments::new_substitutions(
```

Comment:
If the goal is to simplify tests and we are willing to add test specific constructors (or a helper method calling builder), couldn't this all be simplified so instead of this method the test macro call directly looks like:

```
create_rewrite_test! {
test_feature_flag_system_1_treated:
    "test-resources/java/feature_flag_system_1/treated",
    substitutions! {
      "stale_flag_name" => "STALE_FLAG",
      "treated"=>  "true",
      "treated_complement" => "false",
      "namespace" => "some_long_name"
    },
    2
}
```

And the arguments are configured from convention? I.e. language is set based on the path including `/java/` and `input` and `configurations` are simply the default names of the subdirs. Same for figuring out the path to `/expected/`.

It also makes things more readable, because, as it stands now, to understand what:

```
create_rewrite_test! {
    test_feature_flag_system_1_treated: feature_flag_system_1_treated(), "test-resources/java/feature_flag_system_1/treated/expected",2,
}
```

is testing, I actually have to scroll back all the way up to this method. So the tests are less readable than before. And vice-versa, when reading this function, I have no clue how many matches are expected.

If this is not possible, given the differences between the test cases, then I'd still argue that moving half the logic to these calls but then the expected values to the "tabulated" form actually makes these tests harder to follow than the individual test methods.

Reply:
Addressed. Thanks a tonne for simplifying the macro syntax